bot/cogs/tags.py

```diff
@@ -0,0 +1,265 @@
+
+    async def get_tag_data(self, tag_name: Optional[str] = None) -> dict:
+        """
+        Retrieve the tag_data from our API
+
+        :param tag_name: the tag to retrieve
+        :return:
+        if tag_name was provided, returns a dict with tag data.
+        if not, returns a list of dicts with all tag data.
+
+        """
+        headers = {"X-API-KEY": SITE_API_KEY}
+        params = {}
+
+        if tag_name:
+            params['tag_name'] = tag_name
+
+        async with ClientSession() as session:
+            response = await session.get(SITE_API_TAGS_URL, headers=headers, params=params)
+            tag_data = await response.json()
+
+        return tag_data
+
+    async def post_tag_data(self, tag_name: str, tag_content: str) -> dict:
+        """
+        Send some tag_data to our API to have it saved in the database.
+
+        :param tag_name: The name of the tag to create or edit.
+        :param tag_content: The content of the tag.
+        :return: json response from the API in the following format:
+        {
+            'success': bool
+        }
+        """
+
+        headers = {"X-API-KEY": SITE_API_KEY}
+        params = {
+            'tag_name': tag_name,
+            'tag_content': tag_content
+        }
+
+        async with ClientSession() as session:
+            response = await session.post(SITE_API_TAGS_URL, headers=headers, json=params)
+            tag_data = await response.json()
+
+        return tag_data
+
+    async def delete_tag_data(self, tag_name: str) -> dict:
+        """
+        Delete a tag using our API.
+
+        :param tag_name: The name of the tag to delete.
+        :return: json response from the API in the following format:
+        {
+            'success': bool
+        }
+        """
+
+        headers = {"X-API-KEY": SITE_API_KEY}
+        params = {}
+
+        if tag_name:
+            params['tag_name'] = tag_name
+
+        async with ClientSession() as session:
+            response = await session.delete(SITE_API_TAGS_URL, headers=headers, json=params)
+            tag_data = await response.json()
+
+        return tag_data
+
+    @command(name="tags()", aliases=["tags"], hidden=True)
+    async def info_command(self, ctx: Context):
+        """
+        Show available methods for this class.
+
+        :param ctx: Discord message context
+        """
+
+        return await ctx.invoke(self.bot.get_command("help"), "Tags")
+
+    @command(name="tags.get()", aliases=["tags.get", "tags.show()", "tags.show", "get_tag"])
+    async def get_command(self, ctx: Context, tag_name: str = None):
+        """
+        Get a list of all tags or a specified tag.
+
+        :param ctx: Discord message context
+        :param tag_name:
+        If provided, this function shows data for that specific tag.
+        If not provided, this function shows the caller a list of all tags.
+        """
+
+        def _command_on_cooldown(tag_name: Optional[str]) -> bool:
```

Comment:
Why is `tag_name` type hinted as an `Optional`? It doesn't have a default value.